junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,421 @@
+          Throw(exception)
+        }
+
+        val invokeJUnitMethodRhs0 = {
+          methods.zipWithIndex.foldRight[Tree](methodNotFound) {
+            case ((method, id), acc) =>
+              val symbol = method.symbol
+              val headMethodId = Literal(Constant(id.toString))
+              If(gen.mkMethodCall(paramSym, nme.equals_, Nil,
+                List(headMethodId)), gen.mkMethodCall(symbol, Nil),acc)
+          }
+        }
+        val invokeJUnitMethodRhs =
+          atOwner(invokeJUnitMethodSym)(typer.typed(invokeJUnitMethodRhs0))
+        val param = newValDef_compat(paramSym, EmptyTree)()
+        typer.typedDefDef(newDefDef_compat(invokeJUnitMethodSym, invokeJUnitMethodRhs)(
+            vparamss = List(List(param))))
+      }
+
+      def mkGetJUnitMetadataDef(clDef: ClassDef,
+          mlDefOption: Option[ClassDef]): DefDef = {
+        val methods = jUnitAnnotatedMethods(clDef)
+        val mlMethods = mlDefOption.map(jUnitAnnotatedMethods)
+
+        def liftAnnotations(methodSymbol: Symbol): List[Tree] = {
+          val annotations = methodSymbol.annotations
+          def lift(t: Tree): Tree = {
+            t match {
+              case lit: Literal => lit
+
+              case Select(t2, name) => Select(lift(t2), name)
+
+              case ValDef(mod, name, tpt, rhs) =>
+                ValDef(mod, name, tpt, lift(rhs))
+
+              case _ => t
+            }
+          }
+          def liftList(params: List[Tree]): List[Tree] =
+            params.map(lift)
+
+          // Find ad report unsupported JUnit annotations
+          annotations.foreach {
+            case ann if ann.atp =:= testAnnotationType && ann.original.isInstanceOf[Block] =>
+              reporter.error(ann.pos, "@Test(timeout = ...) is not " +
+                "supported in Scala.js JUnit Framework")
+
+            case ann if ann.atp =:= typeOf[org.junit.FixMethodOrder] =>
+              reporter.error(ann.pos, "@FixMethodOrder(...) is not supported " +
+                "in Scala.js JUnit Framework")
+
+            case _ => // all is well
+          }
+
+          // Collect lifted representations of the JUnit annotations
+          annotations.collect {
+            case ann if annotationWhiteList.exists(_ =:= ann.tpe) =>
+              val args = if (ann.args != null) liftList(ann.args) else Nil
+              mkNewInstance(TypeTree(ann.tpe), args)
+          }
+        }
+
+        def defaultMethodMetadata[T: TypeTag](methodAndId: (DefDef, Int)): Tree = {
+          val (m, id) = methodAndId
+          val annotations = liftAnnotations(m.symbol)
+          mkNewInstance[T](List(
+              Literal(Constant(m.name.toString)),
+              Literal(Constant(id.toString)),
+              mkList[java.lang.annotation.Annotation](annotations)))
+        }
+
+        def mkList[T: TypeTag](elems: List[Tree]): Tree = {
+          val array = ArrayValue(TypeTree(typeOf[T]), elems)
```

Comment:
Why not directly creating an `ArrayValue` of type `Array[Object]`? There's no need to have precise type on creation, only to cast it up right afterwards. That also removes the need for the `[T: TypeTag]` altogether.